source/blog/2015-03-22-rom-0-6-0-released.html.markdown

```diff
@@ -0,0 +1,186 @@
+---
+title: ROM 0.6.0 Released
+date: 2015-03-22
+tags: release,announcement
+author: Piotr Solnica
+---
+
+We're happy to announce the final release of ROM 0.6.0! This version brings a lot of improvements and simplifies internal architecture of ROM. There are over 700 commits between 0.5.0 and this release which is a result of fantastic feedback that we've received from the early adopters and many contributions from new people.
+
+This release includes updates of the following gems:
+
+* [rom 0.6.0](https://rubygems.org/gems/rom) - [changelog](https://github.com/rom-rb/rom/blob/master/CHANGELOG.md#v060-2015-03-22)
+* [rom-sql 0.4.0](https://rubygems.org/gems/rom-sql) - [changelog](https://github.com/rom-rb/rom-sql/blob/master/CHANGELOG.md#v040-2015-03-22)
+* [rom-rails 0.3.0](https://rubygems.org/gems/rom-rails) - [changelog](https://github.com/rom-rb/rom-rails/blob/master/CHANGELOG.md#v040-2015-03-22)
+
+Following new adapters have been released too:
+
+* [rom-yesql](https://rubygems.org/gems/rom-yesql)
+* [rom-csv](https://rubygems.org/gems/rom-csv)
+* [rom-yaml](https://rubygems.org/gems/rom-yaml)
+
+## Class-based setup
+
+The biggest change in this release is the ability to define your own relation, mapper and command subclasses that ROM will instantiate for you and expose through its registry. The DSL-based setup is still available but using classes turned out to be a better approach in bigger contexts like web-applications.
+
+Here's a simple Relation subclass definition:
+
+``` ruby
+class Users < ROM::Relation[:memory]
+  def by_name(name)
+    where(name: name)
+  end
+end
+```
+
+## Lazy-relations with data-pipelining
+
+Previous version of ROM had the concept of Reader which turned out to be confusing. After lots of discussions it was replaced by a new interface for accessing relations. This change makes mapping relations explicit as you need to provide the name of mapper(s) that you want to use when loading a relation.
+
+Lazy relations can be used with any object that implements mapper interface but the simplest way to levarage this feature is to register ROM mappers for quick access:
+
+``` ruby
+ROM.setup(:memory)
+
+class Users < ROM::Relation[:memory]
+  def by_name(name)
+    restrict(name: name)
+  end
+end
+
+class UserMapper < ROM::Mapper
+  relation :users
+  register_as :entity
+
+  model User
+
+  attribute :name
+end
+
+rom = ROM.finalize.env
+
+user_entites = rom.relation(:users).as(:entity)
+user_entities.by_name('Jane')
+```
+
+Thanks to data-pipelining feature you can send a relation through multiple mapping objects. This turned out to be very useful for decorating data:
+
+``` ruby
+class UserPresenterMapper < ROM::Mapper
+  relation :users
+  register_as :presenter
+
+  model UserPresenter
+end
+
+user_presenter = rom.relation(:users).as(:entity, :presenter).by_name('Jane').first
```

Comment:
I agree that it’s a good idea to explain the use of both aliases, but for now, I’d recommend sticking to a single alias in all the blog post examples, to keep things consistent.